### ISSUE TITLE
Snakemake can't recognise input directories with relative path

### ISSUE THREAD

**Issue by DrYak** [snakemake/snakemake#456]
**Snakemake version**
 - Used to work with snakemake 5.19.2
 - Fails with the snakemake 5.19.3 (latest currently available in bioconda)

**Describe the bug**
If the input of a rule is a directory, it suddenly only work with absolute paths, otherwise snake make complains about missing input.

**Logs**

(When running with `--dryrun`)

Snakemake 5.19.3 with relative paths:
```
VPIPE_BASEDIR = /Users/localadmin/V-test/V-pipe
Building DAG of jobs...
MissingInputException in line 1 of /Users/localadmin/V-test/V-pipe/rules/visualization.smk:
Missing input files for rule generate_web_visualization:
references/gffs/
```

Snakemake 5.19.2 or using absolute paths in 5.19.3:
```
VPIPE_BASEDIR = /Users/localadmin/V-test/V-pipe
Building DAG of jobs...
Job counts:

  {...job details...}

This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.
```

**Minimal example**
Add a minimal example for reproducing the bug.

This snakemake file only works for 5.19.2:
```python
rule all:
   input:
      gff_dir='references/gffs/'
   output:
      test_file='test.txt'
   shell:
      """
echo "b0rk" > {output.test_file}
      """
```

This config files work for 5.19.3 too:
```python
rule all:
   input:
      gff_dir='/Users/localadmin/V-test/work/references/gffs/'
   output:
      test_file='test.txt'
   shell:
      """
echo "b0rk" > {output.test_file}
      """
```

**Full real-world example**

To exactly reproduce our use case:
 - you can follow the [tutorial here](https://cbg-ethz.github.io/V-pipe/tutorial/sars-cov2/)
 - but use either [this branch](https://github.com/DrYak/V-pipe/tree/sars-cov2)
 - or [that branch](https://github.com/cbg-ethz/V-pipe/tree/sars-cov2-dev)
 - use the following config file:

```
[general]
aligner = bwa

[input]
samples_file = samples.tsv
fastq_suffix =
reference = references/NC_045512.2.fasta
gff_directory = references/gffs/
primers_file = references/primers/nCoV-2019.tsv

[output]
snv = True
local = True
global = False

visualization = True

[hmm_align]
leave_msa_temp = true


[coverage_intervals]
coverage=0

[snv]
mem=2048
threads=24
```

 - specific versions of snakemake can be tested with:
```shell
V=5.19.2; conda deactivate; conda env remove -n v-test ; conda create --yes -n v-test -c conda-forge -c bioconda snakemake-minimal="${V}"; conda activate v-test; snakemake -s "../V-pipe/vpipe.snake" --use-conda --dryrun; echo -ne "${V}\t"; snakemake --version
```

**Comment by johanneskoester** [comment]
The problem is the trailing slash. Commonly, it is not needed. As a workaround, just remove it. It triggers a bug here that I will fix now.

**Comment by johanneskoester** [comment]
Fixed in commit aedbf0f. Thank you!
